core/src/main/java/jenkins/model/item_category/Categories.java

```diff
@@ -0,0 +1,52 @@
+package jenkins.model.item_category;
+
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+import org.kohsuke.stapler.export.Flavor;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * It is a logic representation of a set of {@link Category}.
+ *
+ * This class is not thread-safe.
+ */
+@ExportedBean
+public class Categories implements HttpResponse, Serializable {
```

Comment:
Well currently `ItemGroupMixIn.getCategories` is a `public` API method returning `Categories`. So if `Categories` is restricted, then clearly `getCategories` must be as well—otherwise it would be pointless to call it, since you could not use the result. [Here](#r56849097) I had asked whether such a method needed to exist at all, when it has only one known caller, so it could be subject to an inline refactoring.